tokio/src/sync/semaphore.rs

```diff
@@ -210,6 +210,55 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// ## Ensure Tests with Mutual Dependencies Run Sequentially
+///
+/// By default, Rust runs tests concurrently. However, in some scenarios, concurrent execution might lead to test interference.
```

Comment:
A few things I'd like to change here:

 * The tests actually run in parallel, so I think it's better to say that "Rust runs tests in parallel".
 * This only happens for tests in the same file. I think it would be nice to include that note.

Reply:
done.